src/monexamine.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+#ifndef MONEXAMINE
+#define MONEXAMINE
+
+class monster;
+
+namespace monexamine
+{
+void milk_source( monster *source_mon );
```

Comment:
You shouldn't make the pointer needed, it should be obvious from context that you can sometimes supply `null` there.
But you only use it on known monsters. In fact, the `pet_menu` will crash if you supply it with a `null` monster.

Turning pointers to references and back is easy and readable, keeping things as pointers everywhere only obfuscates the intent.